src/test/regress/expected/non_super_user_object_metadata.out

```diff
@@ -0,0 +1,303 @@
+SET citus.enable_ddl_propagation TO OFF;
+CREATE SCHEMA local_schema;
+SET search_path TO local_schema;
+-- Create type and function that depends on it
+CREATE TYPE test_type AS (f1 int, f2 text);
+CREATE FUNCTION test_function(int) RETURNS test_type
+    AS $$ SELECT $1, CAST($1 AS text) || ' is text' $$
+    LANGUAGE SQL;
+-- Create various objects
+CREATE TYPE mood AS ENUM ('sad', 'ok', 'happy');
+CREATE SEQUENCE test_sequence;
+-- show that none of the objects above are marked as distributed
+SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object where objid = 'local_schema'::regnamespace::oid;
+ pg_identify_object_as_address 
+-------------------------------
+(0 rows)
+
+SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object where objid = 'local_schema.mood'::regtype::oid;
+ pg_identify_object_as_address 
+-------------------------------
+(0 rows)
+
+SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object where objid = 'local_schema.test_type'::regtype::oid;
+ pg_identify_object_as_address 
+-------------------------------
+(0 rows)
+
+SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object where objid = 'local_schema.test_sequence'::regclass::oid;
+ pg_identify_object_as_address 
+-------------------------------
+(0 rows)
+
+SELECT pg_identify_object_as_address(classid, objid, objsubid) from citus.pg_dist_object where objid = 'local_schema.test_function'::regproc::oid;
+ pg_identify_object_as_address 
+-------------------------------
+(0 rows)
+
+SET client_min_messages TO ERROR;
+CREATE USER test_user;
+SELECT run_command_on_workers($$CREATE USER test_user;$$);
+      run_command_on_workers       
+-----------------------------------
+ (localhost,57637,t,"CREATE ROLE")
+ (localhost,57638,t,"CREATE ROLE")
+(2 rows)
+
+RESET client_min_messages;
+GRANT ALL ON SCHEMA local_schema TO test_user;
+SET ROLE test_user;
+SET search_path TO local_schema;
+CREATE TABLE dist_table(a int, b mood, c test_type, d int DEFAULT nextval('test_sequence'), e bigserial);
+-- Citus requires that user must own the dependent sequence
+-- https://github.com/citusdata/citus/issues/5494
+SELECT create_distributed_table('local_schema.dist_table', 'a');
+ERROR:  must be owner of sequence test_sequence
+-- Citus requires that user must own the function to distribute
+SELECT create_distributed_function('test_function(int)');
+ERROR:  must be owner of function test_function
+RESET ROLE;
+SET search_path TO local_schema;
+ALTER SEQUENCE test_sequence OWNER TO test_user;
+ALTER FUNCTION test_function(int) OWNER TO test_user;
+SELECT start_metadata_sync_to_node('localhost', :worker_1_port);
```

Comment:
Can we sync the metadata at the top of the file before even creating anything? We are going to make all tests MX by default soon.

Also, it'd be nice to sync to both nodes as we are going to do that anyway.